packages/sdk/python/human-protocol-sdk/test/human_protocol_sdk/storage/test_storage.py

```diff
@@ -1,219 +1,242 @@
-            download_mock.assert_called_once_with(key=file_key, public=True)
-
-    def test_download_from_public_resource(self):
-        """Download content from public URI"""
-        file_key = "https://s3aaa.com"
-        sample_data = '{"a": 1, "b": 2}'
-
-        with patch("urllib.request.urlopen") as mock_urlopen:
-            cm = MagicMock()
-            cm.read.side_effect = [
-                crypto.encrypt(self.pub_key, sample_data),
-                sample_data.encode("utf-8"),
-            ]
-            mock_urlopen.return_value = cm
-
-            with patch(
-                "human_protocol_sdk.storage.download_from_storage"
-            ) as download_mock:
-                downloaded = download(key=file_key, private_key=self.priv_key)
-                self.assertEqual(json.dumps(downloaded), sample_data)
-                mock_urlopen.assert_called_once()
-
-
-if __name__ == "__main__":
-    unittest.main(exit=True)
+        client.client.bucket_exists = MagicMock(side_effect=[True])
+        result = client.bucket_exists(bucket=self.bucket)
+        self.assertEqual(result, True)
+
+    def test_bucket_not_exists(self):
+        self.client.client.bucket_exists = MagicMock(side_effect=[False])
+        result = self.client.bucket_exists(bucket=self.bucket)
+        self.assertEqual(result, False)
+
+    def test_bucket_error(self):
+        self.client.client.bucket_exists = MagicMock(
+            side_effect=Exception("Connection error")
+        )
+        with self.assertRaises(StorageClientError):
+            self.client.bucket_exists(bucket=self.bucket)
+
+    def test_list_objects(self):
+        file1 = types.SimpleNamespace()
+        file2 = types.SimpleNamespace()
+        file1._object_name = "file1"
+        file2._object_name = "file2"
+        self.client.client.list_objects = MagicMock(side_effect=[[file1, file2]])
+        result = self.client.list_objects(bucket=self.bucket)
+        self.assertEqual(result, ["file1", "file2"])
+
+    def test_list_objects_anonymous(self):
+        client = StorageClient(
+            endpoint_url=self.endpoint_url,
+        )
+        file1 = types.SimpleNamespace()
+        file2 = types.SimpleNamespace()
+        file1._object_name = "file1"
+        file2._object_name = "file2"
+        client.client.list_objects = MagicMock(side_effect=[[file1, file2]])
+        result = client.list_objects(bucket=self.bucket)
+        self.assertEqual(result, ["file1", "file2"])
+
+    def test_list_objects_empty(self):
+        self.client.client.list_objects = MagicMock(side_effect=[[]])
+        result = self.client.list_objects(bucket=self.bucket)
+        self.assertEqual(result, [])
+
+    def test_list_objects_error(self):
+        self.client.client.head_bucket = MagicMock(
+            side_effect=Exception("Connection error")
+        )
+        with self.assertRaises(StorageClientError):
+            self.client.list_objects(bucket=self.bucket)
```

Comment:
----

I added a test to check the length of objects passed is well returned. 
```

def test_list_objects_length(self):
        expected_length = random.randint(1, 10)
        mock_client = MagicMock()
        mock_client.list_objects.return_value = [types.SimpleNamespace(_object_name=f"file{i}") for i in range(expected_length)]
        with patch('human_protocol_sdk.storage.Minio', return_value=mock_client):
            client = StorageClient(endpoint_url='https://example.com', credentials=None)
            object_list = client.list_objects(bucket='my-bucket')
            self.assertEqual(len(object_list), expected_length)
            
    def test_list_objects_length_error(self):
        expected_length = random.randint(1, 10)
        mock_client = MagicMock()
        mock_client.list_objects.return_value = [types.SimpleNamespace(_object_name=f"file{i}") for i in range(expected_length)]
        with patch('human_protocol_sdk.storage.Minio', return_value=mock_client):
            client = StorageClient(endpoint_url='https://example.com', credentials=None)
            object_list = client.list_objects(bucket='my-bucket')
            if len(object_list) != expected_length:
                raise AssertionError(f"Expected {expected_length} objects, but found {len(object_list)}")

```

Reply:
The list_objects function does not return the number of objects in the block, it directly returns a list of objects. Instead, there is another test that checks if the length and the values returned by the function are correct.